### PR TITLE
make default formats of `include` consistent ("sdist" only)…

### PR DESCRIPTION
**Pull request by radoering**
… and refactor code for better maintainability

Related to: python-poetry/poetry#9691
Requires: python-poetry/poetry#9754

- [x] Added **tests** for changed code.
- [x] Updated **documentation** for changed code. (downstream)

### CODE REVIEW

**Comment by Secrus** [comment]
I would still go with `include` being default only for `sdist`. It is most often used to include some extra files like `docs` or `tests` directories and those should not be placed in `site-packages`, as it would happen when included in wheels. Inclusion in wheels should be explicit with a note in docs about how it works and what happens when wheel format is used.

**Comment by radoering** [comment]
> I would still go with `include` being default only for `sdist`.

OK, let's do it. I added a commit to change the default.

File: src/poetry/core/masonry/builders/builder.py [hunk 1]
```diff
@@ -52,41 +53,27 @@ def __init__(
     def _module(self) -> Module:
         from poetry.core.masonry.utils.module import Module
 
-        packages = []
-        for p in self._package.packages:
-            formats = p.get("format") or None
+        packages: list[dict[str, Any]] = []
```
[the remark this answers is not in the document]
Comment (radoering) [reply src/poetry/core/masonry/builders/builder.py:56]: Yes, this should be correct and it seems to work. I can change it if you like.

File: src/poetry/core/masonry/builders/builder.py [hunk 2]
```diff
@@ -52,41 +53,27 @@ def __init__(
...
-                formats
-                and self.format
-                and self.format not in formats
-                and not self._ignore_packages_formats
-            ):
-                continue
+                if (
+                    self.format
+                    and self.format not in formats
+                    and not self._ignore_packages_formats
```
[the remark this answers is not in the document]
Comment (radoering) [reply src/poetry/core/masonry/builders/builder.py:72]: No, but there was a discussion some time ago. See https://github.com/python-poetry/poetry-core/pull/274#discussion_r862134395 and the following comments.

File: src/poetry/core/masonry/builders/builder.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Secrus) [reply src/poetry/core/masonry/builders/builder.py:56]: Yes, please. It's better than `Any`.

File: src/poetry/core/masonry/builders/builder.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Secrus) [reply src/poetry/core/masonry/builders/builder.py:72]: Yeah, let's remove it and make the code simpler to comprehend.

**Review by Secrus — APPROVED** [review]

**Pull request merged**